app/code/core/Mage/Core/Model/Store.php

```diff
@@ -450,12 +450,12 @@ public function setWebsite(Mage_Core_Model_Website $website)
     /**
      * Retrieve store website
      *
-     * @return Mage_Core_Model_Website|false
+     * @return Mage_Core_Model_Website|null
      */
     public function getWebsite()
     {
         if (is_null($this->getWebsiteId())) {
-            return false;
+            return null;
```

Comment:
Backcompatibility breaking change. This can cause huge problems inside openmage itself or many 3rd party modules. From my view, just fix the file `app/code/core/Mage/Customer/sql/customer_setup/mysql4-data-upgrade-1.4.0.0.7-1.4.0.0.8.php` alone

Reply:
Ok, thanks for your advice.